src/main/java/Duke.java

```diff
@@ -1,10 +1,119 @@
+
+    /**
+     *
+     * List of all Tasks
+     */
+    private static Task[] allTasks;
+
+    private static String input;
+    private static String logo = getLogo();
+    private static String line = getLine();
+    private static int inputCount = 0;
+
     public static void main(String[] args) {
+        initTaskList();
+        welcomeMessage(logo, line);
+        while(true) {
+            String userInput = getUserInput();
+            taskQuery(userInput);
+        }
+    }
+
+    private static String getUserInput() {
+        input = in.nextLine();
+        System.out.println(line);
+        return input;
+    }
+
+    private static void taskQuery(String userInput) {
+        if(userInput.equals("bye")) {
+            System.out.println("Bye, see you!\n" + line);
+            System.exit(0);
+        }
+        else if(userInput.equals("list")) {
+            for(int i = 0; i < inputCount; i++){
+                //System.out.println((i+1) + ". [" + inputs[i].getStatusIcon() + "] " +
+                System.out.println((i+1) + "." + allTasks[i]);
+            }
+        }
+        //assumes user will follow format "done x"
+        else if(userInput.startsWith("done ")) {
+            String[] words = userInput.split(" ");
+            int taskCompleted = Integer.parseInt(words[1]) - 1;
+            if (allTasks[taskCompleted].isDone()) {
+                System.out.println("You have already marked this task as done! Time to move on :)");
+            }
+            else {
+                allTasks[taskCompleted].markAsDone();
+                System.out.println("Awesome! You've completed the following task:");
+                System.out.println(" [X] " + allTasks[taskCompleted].getDescription());
+            }
+
+        }
+        else if(userInput.startsWith("todo ")) {
+            String taskName = userInput.substring(5);
+            ToDo newToDo = new ToDo(taskName);
+            allTasks[inputCount] = newToDo;
+            inputCount += 1;
+            taskMessage(inputCount, newToDo);
+        }
+        else if(userInput.startsWith("deadline ") && userInput.contains("/by")) {
+            int taskEndIndex = userInput.indexOf("/by") - 1; //to account for spacing before "/by"
+            String taskName = userInput.substring(9, taskEndIndex);
+            int deadlineStartIndex = taskEndIndex + 5;
+            String deadline = userInput.substring(deadlineStartIndex);
+            Deadline newDeadline = new Deadline(taskName, deadline);
+            allTasks[inputCount] = newDeadline;
+            inputCount += 1;
+            taskMessage(inputCount, newDeadline);
+        }
+        else if(userInput.startsWith("event ") && userInput.contains("/at")) {
+            int taskEndIndex = userInput.indexOf("/at") - 1; //to account for spacing before "/by"
+            String taskName = userInput.substring(6, taskEndIndex);
+            int eventTimeStartIndex = taskEndIndex + 5;
```

Comment:
can consider storing the numbers as static final variables so we know how they originate

Reply:
Agree with Zhen Hong. Do no use magic numbers. For example in words[1] instead of using 1, use a constant such as TASK_NAME or DATE to describe it